lib/utils.rb

```diff
@@ -0,0 +1,7 @@
+module Utils
+  module_function
+
+  def generate_id(hash = {}, keys = [])
+    Digest::SHA1.hexdigest(keys.map { |k| hash[k] }.join)
+  end
+end
```

Comment:
Actually I joined the Rubymine tripe recently, but this one escaped the checking somehow :-P